### PR TITLE
[student-libraries] Give teachers the ability to share libraries with their classes

### PR DESCRIPTION
**Pull request by jmkulwik**
This provides a modal so teachers can selectively share libraries with the students in their classes.

New Functionality:
ShareLibrariesWithClasses

Mockup (for reference)
Slice 1 (17)

## Links

- [spec](https://docs.google.com/document/d/1zZ6WU5oBfh5Ra5TkdUpYRFsoMpIu9MRCWO42oBMyhrM/edit#)
- [jira](https://codedotorg.atlassian.net/browse/STAR-777)
- some prefactoring for this task: https://codedotorg.atlassian.net/browse/STAR-777

## Testing story
Added unit tests

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/Button.jsx [hunk 1]
```diff
@@ -203,7 +205,8 @@ class Button extends React.Component {
       id,
       tabIndex,
       isPending,
-      pendingText
+      pendingText,
+      useButtonTag
```
Comment (jmkulwik) [inline apps/src/templates/Button.jsx:209]: This is temporary. I'm currently working on deprecating the `<div>` and `<a>` version of the Button.

File: apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx [hunk 2]
```diff
@@ -104,16 +107,27 @@ class LibraryCreationDialog extends React.Component {
...
           this.setState({dialogState: DialogState.UNPUBLISHED})
         }
+        onShareTeacherLibrary={this.onShareTeacherLibrary()}
         libraryDetails={libraryDetails}
         libraryClientApi={libraryClientApi}
       />
     );
   };
 
+  hasTeacherSections() {
```
Comment (maddiedierker) [inline apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx]: nit: this function should be named `isTeacher()`

File: apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx [hunk 3]
```diff
@@ -123,6 +137,8 @@ class LibraryCreationDialog extends React.Component {
           <PublishSuccessDisplay
             libraryName={libraryName}
             channelId={channelId}
+            // Waiting for design guidance prior to enabling this functionality
+            // onShareTeacherLibrary={this.onShareTeacherLibrary()}
```
Comment (maddiedierker) [inline apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx]: i know this is commented out, but just a heads up that the parentheses will need to be removed here:
```suggestion
            // onShareTeacherLibrary={this.onShareTeacherLibrary}
```

File: apps/src/templates/Button.jsx [hunk 4]
```diff
@@ -213,12 +216,17 @@ class Button extends React.Component {
       throw new Error('Expect at least one of href/onClick');
     }
 
-    const Tag = href ? 'a' : 'div';
+    let Tag = href ? 'a' : 'div';
+    Tag = useButtonTag ? 'button' : Tag;
```
Comment (maddiedierker) [inline apps/src/templates/Button.jsx:220]: nit: can we use if/elsif/else here instead of 2 ternaries?

**Review by maddiedierker — APPROVED** [review]
left a couple of small nit comments, but none are merge-blockers. 

LGTM! this looks awesome, jessie 🎉

File: apps/src/templates/Button.jsx (hunk 4 above)
Comment (jmkulwik) [reply apps/src/templates/Button.jsx:220]: I'll go ahead and address this in the button PR. https://github.com/code-dot-org/code-dot-org/pull/33609

**Pull request merged**